### ISSUE TITLE
ucore-hci - Add LXC Container Management via Incus

### ISSUE THREAD

**Issue by cmspam** [ublue-os/ucore#217]
**Feature Request: Add LXC Container Management via Incus**

I’m very happy that `ucore-hci` for virtual machines, but I’d love to see support for managing LXC containers as well.

`Incus` (https://github.com/lxc/incus) is a fork of LXD that excels at managing both LXC containers and virtual machines. While I currently maintain a containerized solution (https://github.com/cmspam/incus-docker) for running Incus, I’ve always envisioned having an immutable appliance with built-in Incus support. Given the existing ZFS support in `ucore-hci`, Incus could be a great addition to this project.

Incus has official Fedora packages, but more up-to-date packages are available via Copr (https://copr.fedorainfracloud.org/coprs/ganto/lxc4/).

Would there be any interest in exploring this integration?

**Comment by m2Giles** [comment]
We currently offer a distrobox with incus inside of it. It works when used in rootful mode.

**Comment by cmspam** [comment]
Indeed, incus can certainly run containerized, but I think there is a big advantage to having it available on a base image like this one, dedicated to virtualization uses, as opposed to having to go through a distrobox or a podman container. Especially seeing as that incus now runs OCI containers, I think it's one of the use cases where it makes more sense to have it be one of the few things not containerized.

**Comment by m2Giles** [comment]
Yes, I have it on my Ucore image right now.

The big hesitation is that clustered incus instances need to be updated in lockstep. This is not an issue for single instance configurations.

Ultimately, sysext looks like it will be a great solution for this.

**Comment by cmspam** [comment]
I actually hadn't thought about that until your comment, but I can understand how the full OS image installation and reboots could certainly complicate things for clusters.
